### DISCUSSION TITLE
Syntax requirements? (and potential typo)

### DISCUSSION THREAD

**Discussion by mojofawad** [vercel/next.js#40243] [Help]
I'm going through the learning documentation and came across two instances of things that gave me pause. One is a simple typo, the other is a question regarding syntax requirements?

---

### Typo

At the bottom of [this page](https://nextjs.org/learn/basics/navigate-between-pages/link-component), I believe it should say "not" rather than "nor". 

<img width="413" alt="image" src="https://user-images.githubusercontent.com/54484850/188547928-b6017e37-52c9-4efb-9c85-ae6dac91125c.png">

---

### Syntax: Semicolons required?

I'm wondering about the use of semicolons and their requirements. 

While following the guide, I came across this and noticed that the example files provided did not have a semicolon at the end. Yet, when told to add an import, the follow along code ends the import statement with a semicolon. 

_From the guide:_ [Link to page](https://nextjs.org/learn/basics/navigate-between-pages/link-component)

<img width="494" alt="image" src="https://user-images.githubusercontent.com/54484850/188548372-cffed5c7-8bcc-4380-853e-699bb6268e6d.png">

_Imported code from guide: (I added Line 2, but Line 1 was already there)_

<img width="504" alt="image" src="https://user-images.githubusercontent.com/54484850/188548851-1aa66696-cf53-48f3-9968-75996b285f3d.png">

I'm sure this isn't a big deal, but I'm just confirming so I can be consistent going forward.

**Comment by icyJoseph** [comment]
I think for the typo, perhaps you should open a PR to fix it.

---

Regarding usage of semi-colons, in general that's a property of the language. Meaning that, while the teaching documentation should be consistent, the JavaScript language doesn't really care about it. Technically, the interpreter fills in the semi-colons when parsing the code, IIRC. Though this has a few pitfalls one should be aware of.

Moreover, the usage of automated code formatters, keeps most from thinking about the semi-colon. At the end of the day, it depends on the preferences of those running the project.

Personally, I use semi-colons, but if you are getting started with JS, I'd recommend this video, https://www.youtube.com/watch?v=cB0STl_A4l4. 

Again, **I do agree that the documentation should be consistent in the usage of semi-colons.**
